clang/lib/DPCT/APINamesCUB.inc

```diff
@@ -617,192 +617,356 @@ CONDITIONAL_FACTORY_ENTRY(
                     Diagnostics::REDUCE_PERFORMANCE_TUNE))))))
 
 // cub::DeviceRadixSort::SortKeys
-CONDITIONAL_FACTORY_ENTRY(
-    CheckCubRedundantFunctionCall(),
-    REMOVE_API_FACTORY_ENTRY("cub::DeviceRadixSort::SortKeys"),
-    REMOVE_CUB_TEMP_STORAGE_FACTORY(FEATURE_REQUEST_FACTORY(
-        HelperFeatureEnum::DplExtrasAlgorithm_sort_keys,
-        HEADER_INSERT_FACTORY(
-            HeaderType::HT_DPCT_DPL_Utils,
-            REMOVE_CUB_TEMP_STORAGE_FACTORY(CONDITIONAL_FACTORY_ENTRY(
-                makeCheckAnd(CheckArgCount(8, std::greater_equal<>(),
-                                           /* IncludeDefaultArg */ false),
-                             makeCheckNot(CheckArgIsDefaultCudaStream(7))),
-                CALL_FACTORY_ENTRY(
-                    "cub::DeviceRadixSort::SortKeys",
-                    CALL("dpct::sort_keys",
-                         CALL("oneapi::dpl::execution::device_policy",
-                              STREAM(7)),
-                         ARG(2), ARG(3), ARG(4), LITERAL("false"), ARG(5),
-                         ARG(6))),
-                CONDITIONAL_FACTORY_ENTRY(
-                    CheckArgCount(7, std::greater_equal<>(),
-                                  /* IncludeDefaultArg */ false),
-                    CALL_FACTORY_ENTRY(
-                        "cub::DeviceRadixSort::SortKeys",
-                        CALL("dpct::sort_keys",
+CASE_FACTORY_ENTRY(
+  CASE(CheckCubRedundantFunctionCall(),
+       REMOVE_API_FACTORY_ENTRY("cub::DeviceRadixSort::SortKeys")),
+  OTHERWISE(
+       FEATURE_REQUEST_FACTORY(
+       HelperFeatureEnum::DplExtrasAlgorithm_sort_keys,
+       HEADER_INSERT_FACTORY(
+       HeaderType::HT_DPCT_DPL_Utils,
+       REMOVE_CUB_TEMP_STORAGE_FACTORY(
+       CASE_FACTORY_ENTRY(
```

Comment:
Seems like  these two macro `CASE_FACTORY_ENTRY`  `IFELSE_FACTORY_ENTRY` are about the same?

Reply:
Yea, case factory is essentially just a chain of if-else factories. I added it because the deep nesting of the if-else factories was getting hard to read.